docs/en/advanced_guides/evaluator.md

```diff
@@ -1,5 +1,220 @@
+        different. For generative models, different metric need image
+        generated with different inputs. For example, FID, KID and IS need
+        images generated with random noise, and PPL need paired images on the
+        specific noise interpolation path. Therefore, we first group metrics
+        with respect to their sampler's mode (refers to
+        :attr:~`GenMetrics.SAMPLER_MODE`), and build a shared sampler for each
+        metric group. To be noted that, the length of the shared sampler
+        depends on the metric of the most images required in each group.
+
+        Args:
+            module (BaseModel): Model to evaluate. Some metrics (e.g. PPL)
+                require `module` in their sampler.
+            dataloader (DataLoader): The dataloader for real image.
+
+        Returns:
+            List[Tuple[List[BaseMetric], Iterator]]: A list of "metrics-shared
+                sampler" pair.
+        """
+        if self.metrics is None:
+            return [[[None], []]]
+
+        # grouping metrics based on `SAMPLER_MODE` and `sample_mode`
+        metric_mode_dict = defaultdict(list)
+        for metric in self.metrics:  # Specify a sampler group for each metric.
+            metric_md5 = self._cal_metric_hash(metric)
+            metric_mode_dict[metric_md5].append(metric)
+
+        metrics_sampler_list = []
+        for metrics in metric_mode_dict.values(): # Generate a sampler for each group.
+            first_metric = metrics[0]
+            metrics_sampler_list.append([
+                metrics,
+                first_metric.get_metric_sampler(module, dataloader, metrics)
+            ])
+
+        return metrics_sampler_list
+```
+
+The method will first check if it has any evaluation metrics to calculate: if not, it will return directly. If there are metrics to calculate, it will iterate through all the evaluation metrics and group them based on the sampler_mode and sample_model. The specific implementation is as follows: it calculates a hash code based on the sampler_mode and sample_model, and puts the evaluation metrics with the same hash code into the same list.
+
+```python
+class Evaluator(Evaluator):
+	...
+    @staticmethod
+    def _cal_metric_hash(metric: GenMetric):
+        """Calculate a unique hash value based on the `SAMPLER_MODE` and
+        `sample_model`."""
+        sampler_mode = metric.SAMPLER_MODE
+        sample_model = metric.sample_model
+        metric_dict = {
+            'SAMPLER_MODE': sampler_mode,
+            'sample_model': sample_model
+        }
+        if hasattr(metric, 'need_cond_input'):
+            metric_dict['need_cond_input'] = metric.need_cond_input
+        md5 = hashlib.md5(repr(metric_dict).encode('utf-8')).hexdigest()
+        return md5
+```
+
+Finally, this method will generate a sampler for each evaluation metric group and add it to a list to return.
+
+## Evaluation process of an evaluator
+
```

Comment:
The implementation of evaluation process can be found in `mmagic.engine.runner.MultiValLoop.run` and `mmagic.engine.runner.MultiTestLoop.run`. Here we take `mmagic.engine.runner.MultiValLoop.run` as example:

Reply:
Thanks for your comments, I have added a paragraph for more clear explaination